### PR TITLE
[JSDoc] Add missing documentation for botbuilder-ai

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 2602

## Description
This PR introduces the change to `jsdoc/require-jsdoc` in the [.eslintrc.json](https://github.com/microsoft/botbuilder-js/blob/master/.eslintrc.json#L55) file from `warn` to `error` and adds all the missing JSDoc documentation for [botbuilder-ai](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botbuilder-ai) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

## Specific Changes

  - Updated `jsdoc/require-jsdoc` in `.eslintrc.json` from `warn` to `error`.
  - Private methods' documentation that does not exist on botbuilder-dotnet we added the `@private` tag in the JSDoc comment.
  - Added JSDoc comments in **10 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
imagen

### CODE REVIEW

File: libraries/botbuilder-ai/src/qnamaker-utils/activeLearningUtils.ts [hunk 1]
```diff
@@ -65,7 +65,9 @@ export class ActiveLearningUtils {
         return filteredQnaSearchResult;
     }
 
-
+    /**
+     * @orivate
```
Comment (ceciliaavila) [inline libraries/botbuilder-ai/src/qnamaker-utils/activeLearningUtils.ts:69]:
```suggestion
     * @private
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-ai/src/qnamaker-utils/activeLearningUtils.ts (hunk 1 above)
Comment (sw-joelmut) [reply libraries/botbuilder-ai/src/qnamaker-utils/activeLearningUtils.ts:69]: 👀

Comment (sw-joelmut) [reply libraries/botbuilder-ai/src/qnamaker-utils/activeLearningUtils.ts]: Fixed, thank you.

**Review by ceciliaavila — APPROVED** [review]

File: libraries/botbuilder-ai/src/luisRecognizer.ts [hunk 2]
```diff
@@ -265,13 +265,30 @@ export class LuisRecognizer implements LuisRecognizerTelemetryClient {
 
     /**
      * Creates a new LuisRecognizer instance.
-     * @param application An object conforming to the [LuisApplication](#luisapplication) definition or a string representing a LUIS application endpoint, usually retrieved from https://luis.ai.
-     * @param options (Optional) options object used to control predictions. Should conform to the [LuisRecognizerOptions](#luisrecognizeroptions) definition.
-     * @param includeApiResults (Deprecated) flag that if set to `true` will force the inclusion of LUIS Api call in results returned by [recognize()](#recognize). Defaults to a value of `false`.
+     * @param application The LUIS application endpoint, usually retrieved from https://luis.ai.
+     * @param options (Optional) Options object used to control predictions. Should conform to the [LuisRecognizerOptions](#luisrecognizeroptions) definition.
+     * @param includeApiResults (Deprecated) Flag that if set to `true` will force the inclusion of LUIS Api call in results returned by [recognize()](#recognize). Defaults to a value of `false`.
```
Comment (santgr11) [inline libraries/botbuilder-ai/src/luisRecognizer.ts:270]: When testing this kind of reference in vs code I get an error. I don't know if this is the proper way to link to reference other classes.
image

Comment (sw-joelmut) [reply libraries/botbuilder-ai/src/luisRecognizer.ts:270]: Yea, maybe I could remove the link reference, I replicated it from the previous comments

Comment (sw-joelmut) [reply libraries/botbuilder-ai/src/luisRecognizer.ts]: Resolved

**Review by santgr11 — APPROVED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**